modules/build/src/test/scala/scala/build/tests/BuildOptionsTests.scala

```diff
@@ -19,6 +25,167 @@ class BuildOptionsTests extends munit.FunSuite {
     )
   }
 
+  test("-S 3.nightly option works") {
+    val options = BuildOptions(
+      scalaOptions = ScalaOptions(
+        scalaVersion = Some("3.nightly"),
+        scalaBinaryVersion = None,
+        supportedScalaVersionsUrl = None
```

Comment:
The `None` are the default values, no need to specify them. Same for most `BuildOptions` in this file.

```suggestion
        scalaVersion = Some("3.nightly")
```